helix-core/src/main/java/org/apache/helix/util/HelixUtil.java

```diff
@@ -152,6 +152,18 @@ public static String serializeByComma(List<String> objects) {
     return keyValueMap;
   }
 
+  /**
+   * Performs conversion from a map string into a map. The string was converted by map's toString().
+   *
+   * @param mapAsString A string that is converted by map's toString() method.
+   *                    Example: "{k1=v1, k2=v2}"
+   * @return Map<String, String>
+   */
+  public static Map<String, String> stringToMap(String mapAsString) {
```

Comment:
If you encode the original entry map using Json, then you don't need this method here which looks very hacky.